picard/webservice.py

```diff
@@ -246,17 +297,32 @@ def url_port(url):
             return url.port(443)
         return url.port(80)
 
-    def _handle_reply(self, reply, request, handler, xml, refresh):
+    def _handle_reply(self, reply, wsrequest):
+        handler = wsrequest.handler
         error = int(reply.error())
         if error:
-            log.error("Network request error for %s: %s (QT code %d, HTTP code %s)",
-                      reply.request().url().toString(QUrl.RemoveUserInfo),
-                      reply.errorString(),
-                      error,
-                      repr(reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute))
-                      )
-            if handler is not None:
+            code = int(reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute))
+            errstr = reply.errorString()
+            url = reply.request().url().toString(QUrl.RemoveUserInfo)
+            log.error("Network request error for %s: %s (QT code %d, HTTP code %d)",
+                      url,errstr, error, code)
+
+            if (wsrequest.retries < TEMP_ERRORS_RETRIES
+                and (code == 503
+                     or code == 429
```

Comment:
It isn't documented, because it isn't used yet, but web service returns 503 for historical reasons, and since 429 appeared and should be used instead.
Not mandatory to support it now, though it doesn't hurt.